### PR TITLE
addpkg: libredefender

### PR DESCRIPTION
**Pull request by Ast-x64**

### CODE REVIEW

File: libredefender/riscv64.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
++        'SKIP'
++        'cf82f590fcdf3b98c62a8494e685d8e1e53a7009e5361be91e8c480a55af7040dbbdf2ede22f49a5cbb027be37b75e225a7da37a4110d2c4d3165bb51ef75ddd'
++        '9226d72280a7b408575cb6789193ed61f1861e5c465510a7b1bf555e238f482e73b645d50d3b5d03dd6d7afe96fb975ba73ef6fd5accca3df0d2204c0ea75a6d')
+ 
+ validpgpkeys=("64B13F7117D6E07D661BBCE0FE763A64F5E54FD6")
+ 
+ prepare() {
+   cd "${pkgname}-${pkgver}"
+-  cargo fetch --locked --target "$CARCH-unknown-linux-gnu"
++  cargo fetch --locked --target "riscv64gc-unknown-linux-gnu"
```
Comment (r-value) [inline libredefender/riscv64.patch:24]: I don't think it's a good idea to substitute the variables with specific values, which reduces the generality of `PKGBUILD`.

File: libredefender/riscv64.patch [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
++  cd ~/.cargo/registry/src/github.com-1ecc6299db9ec823/nix-0.17.0/src/sys/ptrace
++  patch -Np1 -i "${srcdir}/fix_ptrace_enum.patch"
```
Comment (r-value) [inline libredefender/riscv64.patch:26]: `nix-0.23.0`is available and compatible with RISC-V, please remove the patch and bump the dependency instead.

File: libredefender/fix_type.patch [hunk 3]
```diff
@@ -0,0 +1,40 @@
+diff --git a/engine.rs b/engine.rs
+index 5f2fe24..e26bc58 100644
+--- a/engine.rs
++++ b/engine.rs
+@@ -78,7 +78,7 @@ pub struct Engine {
+ unsafe impl Send for Engine {}
+ unsafe impl Sync for Engine {}
+ 
+-fn map_scan_result(result: cl_error_t, virname: *const i8) -> Result<ScanResult, ClamError> {
++fn map_scan_result(result: cl_error_t, virname: *const u8) -> Result<ScanResult, ClamError> {
```
Comment (r-value) [inline libredefender/fix_type.patch:10]: Are these changes compatible with x86_64? I really don't think that it's the source of the problem, could you please explain the reasons for this patch further?

**Review by r-value — CHANGES_REQUESTED** [review]

**Comment by r-value** [comment]
Any updates?

**Comment by Xeonacid** [comment]
Superseded by #2895 , closing.

**Pull request closed without merging**